iree/compiler/Dialect/Flow/IR/test/tensor_folding.mlir

```diff
@@ -20,6 +20,15 @@ func @reshapeNoOpStatic(%arg0: tensor<4x4xf32>) -> tensor<4x4xf32> {
 
 // -----
 
+// CHECK-LABEL: @reshapeRankDifferent
+func @reshapeRankDifferent(%arg0: tensor<1xf32>) -> tensor<f32> {
+  // CHECK-NEXT: flow.tensor.reshape %arg0
+  %0 = flow.tensor.reshape %arg0 : tensor<1xf32> -> tensor<f32>
```

Comment:
Interesting test case. I would think this should be a special case and should be treated as the same. THe number of elements is 1 in the source and 0 in the target, but is really 1 in reality.

@benvanik any thoughts?